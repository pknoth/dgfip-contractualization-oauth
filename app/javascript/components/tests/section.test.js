
/* eslint no-unused-vars: "error" */
import ReactTestRenderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import Section from '../section'

import Container from './../container'

describe('components | Section', () => {
  describe('render', () => {
    const props = {
      title: 'The title',
      imageSrc: '/somepath/image.png',
      altText: 'some alternative text',
      children: '<p>Some html</p>'
    }
    const renderer = ReactTestRenderer.create(<Section {...props} />)
    it('should be defined', () => {
      expect(renderer).toBeDefined()
    })
    it('should match snapshot', () => {
      expect(renderer).toMatchSnapshot()
    })
  })

  describe('Container', () => {
    const props = {
      title: 'The title',
      imageSrc: '/somepath/image.png',
      altText: 'some alternative text',
      children: '<p>Some html</p>'
    }
    it('should have proper title and content', () => {
      const wrapper = shallow(<Section {...props} />)
      const container = wrapper.find(Container)
      expect(
        container
          .find('div')
          .at(0)
          .props().children,
      ).toEqual([<h2>The title</h2>, '<p>Some html</p>'])
    })
    it('should have proper image with alternative texte', () => {
      const wrapper = shallow(<Section {...props} />)
      const container = wrapper.find(Container)
      expect(
        container
          .find('div')
          .at(1)
          .props().children,
      ).toEqual(<img alt={props.altText} src={props.imageSrc} />)
    })
  })
})
