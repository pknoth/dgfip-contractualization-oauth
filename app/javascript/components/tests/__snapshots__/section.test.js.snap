// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components | Section render should match snapshot 1`] = `
<section>
  <div
    className="jsx-1157840685"
    style={
      Object {
        "maxWidth": 1200,
      }
    }
  >
    <div>
      <h2>
        The title
      </h2>
      &lt;p&gt;Some html&lt;/p&gt;
    </div>
    <div>
      <img
        alt="some alternative text"
        src="/somepath/image.png"
      />
    </div>
  </div>
</section>
`
